### DISCUSSION TITLE
fatal: Could not read from remote repository. Git push error [duplicate]

### DISCUSSION THREAD

**Discussion by cagataysen** [community/community#73457] [New to GitHub] [answered]
In the past I could use git bash and git desktop applications. I have decided to use after some time gitbash to improve my skills and trying to push changes on my local to the server.

I have realized I can't push anything even on extensions on my editors and I decided to try with the most basic steps and I'm getting this error.

ssh: Could not resolve hostname github.com:firstcontributions: Name or service not known

fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

I have tried many solutions such as changing it to the from https to ssh git remote set-url origin git@github.com:cagataysen/first-contributions.git but after trying this command: git push --set-upstream origin cagatay-first-contribution I still get git@github.com: Permission denied (publickey). fatal: Could not read from remote repository.

Please make sure you have the correct access rights and the repository exists.

After checking this error message on google: https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey

I followed the steps: To make sure you are connecting to the right domain I texted this command: $ ssh -vT git@github.com

after many debugging messages I got this. git@github.com: Permission denied (publickey).

https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories

I have realized something is wrong with my remote-repository settings and tried this forum but result didn't change.

I removed my account and password from my credential manager on Windows.

I have tried to adding SSH key to my profile via SSH and GPG keys part. While using this instructors: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account

https://docs.github.com/en/authentication/connecting-to-github-with-ssh/working-with-ssh-key-passphrases

I think I can't detect my exact problem at the end of the day and looking for a solution. Thanks indeed.

**Comment by cagataysen** [comment, marked as answer]
Solution worked for me: I had two emails related to my account. After deleting the previous one from my account and entering globally login with my primary email on bash I could push.

git config --global user.email “[valid-email]”

**Reply by aamahi** [reply]
Not working

**Reply by AnandKumarSinha07** [reply]
bro do you able to solve it

**Reply by cagataysen** [reply]
Yes, I realized I have changed my email which is connected to my github account and I had to change the global email on my gitbash account so I basically logout and login with correct email. The problem can be different on your case, but you need to focus on cleaning related account which includes email and password, then login again. Try to find out windows cached email and password and clean the default ones. I wish you good luck cuz it took my 3 days to understand and solve it, but problem is what I talked about focus on this part.

**Comment by mindashu** [comment]
I still getting the same Error even if the single email id is present. Please help with any solution on it
